File: java-store-management-system/src/Store.java
import java.util.ArrayList;
import java.util.Date;
import java.util.UUID;

public class Store {
    private UUID uuid;
    private String address;
    private ArrayList<Employee> employees = new ArrayList<>();
    private ArrayList<Item> items = new ArrayList<>();

    public ArrayList<Item> getItems() {
        return items;
    }

    public Item addItem(Item item) {
        // check if the item exists (by uuid)
        for (Item itemIterator : items) {
            if (itemIterator.getUuid() == item.getUuid()) {
                return null;
            }
        }

        // otherwise, we add the new employee
        items.add(item);

        return item;
    }

    public Store() {
        this.uuid = UUID.randomUUID();
    }

    public Store(String address) {
        this.uuid = UUID.randomUUID();
        this.address = address;
    }

    public UUID getUuid() {
        return uuid;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public ArrayList<Employee> getEmployees() {
        return employees;
    }

    public Employee getEmployee(String SSN) {
        // gets an employee from the employees list by SSN
        for (Employee employeeIterator : employees) {
            if (employeeIterator.getSSN() == SSN) {
                return employeeIterator;
            }
        }

        // if it doesn't exist, return null
        return null;
    }

    // adding an employee to the employees list
    public Employee addEmployee(Employee employee) {
        // check if the employee exists (by SSN)
        for (Employee employeeIterator : employees) {
            if (employeeIterator.getSSN() == employee.getSSN()) {
                return null;
            }
        }

        // otherwise, we add the new employee
        employees.add(employee);

        return employee;
    }

    public void printInformation() {
        String message = String.format("Information on the store with id %s\nAddress: %s\nNumber of employees ",
                uuid, address, employees.size() > 0 ? employees.size() : 0);
        System.out.println(message);
    }
}

File: python-store-management-system/Store.py
import uuid


class Store:
    def __init__(self, address, employees=[], items=[]):
        self.address = address
        self.uuid = uuid.uuid4()
        self.employees = employees
        self.items = items

    def __str__(self):
        return f"Store: {self.uuid}, {self.address}"

    def get_employee(self, SSN):
        for employee in self.employees:
            if employee.SSN == SSN:
                return employee

        return None

    def add_employee(self, employee):
        # check if the employee already exists
        for employee1 in self.employees:
            if employee1.SSN == employee.SSN:
                return None

        self.employees.append(employee)
        return employee

    def get_item(self, uuid):
        for item in self.items:
            if item.uuid == uuid:
                return item

        return None

    def add_item(self, item):
        # check if the item already exists
        for item1 in self.items:
            if item1.uuid == item.uuid:
                return None

        self.items.append(item)
        return item
